### PR TITLE
Daos/books languages

### PR DESCRIPTION
**Pull request by sagar-wednesday**
### Ticket Link

---

### Related Links

---

### Description

---

### Steps to Reproduce / Test

---

---

### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's

---

### CODE REVIEW

File: server/cronJobs/aggregateJob.js [hunk 1]
```diff
@@ -1,12 +1,5 @@
-import { getAllCategories } from '@server/daos/products';
 import moment from 'moment';
-import {
-  getCountByDate,
-  getCountByDateForCategory,
```
Comment (alichherawalla) [inline server/cronJobs/aggregateJob.js:5]: is this file still relevent?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -31,11 +25,6 @@ jobs:
         run: yarn test
       - name: Build
         run: yarn build:local
-      - name: SonarCloud Scan
-        uses: sonarsource/sonarcloud-github-action@master
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```
Comment (alichherawalla) [inline .github/workflows/ci.yml:38]: let's use the latest version of `artiomtr/jest-coverage-report-action` and configure for annotations

File: server/cronJobs/aggregateJob.js (hunk 1 above)
Comment (sagar-wednesday) [reply server/cronJobs/aggregateJob.js:5]: nope.....I removed it

File: server/utils/queue.js [hunk 3]
```diff
@@ -14,25 +12,15 @@ export const QUEUE_PROCESSORS = {
...
   }
 };
 
-const CRON_EXPRESSIONS = {
-  MIDNIGHT: '0 0 * * *'
-};
-
 export const initQueues = () => {
   console.log('init queues');
   Object.keys(QUEUE_PROCESSORS).forEach(queueName => {
```
Comment (alichherawalla) [inline server/utils/queue.js:20]: can we remove this as well

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -1,12 +1,6 @@
 name: Node Express GraphQL Template CI
 
-on:
-  pull_request:
-    branches: 
-      - develop
-  push:
-    branches:
-      - develop
+on: [push, pull_request]
```
Comment (alichherawalla) [inline .github/workflows/ci.yml]: this will run on every push and every pull_request

**Review by sagar-wednesday — COMMENTED** [review]
reviewed it once and isAuthenticated tests are commented and will be removed/updated in future PRs.

File: server/daos/tests/authors.test.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      expect(async () => insertAuthor(authorArgs)).rejects.toThrowError(errorMessage);
+      expect(createAuthorSpy).toHaveBeenCalledWith(authorArgs);
+    });
+  });
+  describe('updateAuthor', () => {
+    let updateAuthorSpy;
+    let expectedArgs;
+    beforeEach(() => {
+      updateAuthorSpy = jest.spyOn(db.authors, 'update');
+      expectedArgs = {
```
Comment (alichherawalla) [inline server/daos/tests/authors.test.js:45]: i like this

File: server/daos/tests/authorsBooks.test.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+      });
+
+      expect(async () => insertAuthorsBooks(authorsBooksArgs)).rejects.toEqual(new Error(errorMessage));
+    });
+  });
+
+  describe('updateAuthorsBooks for books', () => {
+    let authorsBooksForBooksArgs;
+    beforeEach(() => {
+      authorsBooksForBooksArgs = [
```
Comment (alichherawalla) [inline server/daos/tests/authorsBooks.test.js]: this is good, i see how you're reusing things. 

You always want to make sure that your test file is super readable and this is making it quite long. Let's do this 

- create another file in the tests folder called mockData.js
- move these variables with meaningful names in the mockdata file. Just invoke here

File: server/daos/tests/authorsBooks.test.js [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+      });
+
+      expect(async () => updateAuthorsBooksForBooks(authorsBooksForBooksArgs)).rejects.toEqual(new Error(errorMessage));
+    });
+  });
+
+  describe('updateAuthorsBooks for authors', () => {
+    let authorsBooksForAuthorsArgs;
+    beforeEach(() => {
+      authorsBooksForAuthorsArgs = [
```
Comment (alichherawalla) [inline server/daos/tests/authorsBooks.test.js]: same for this

File: server/daos/tests/booksLanguages.test.js [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+      });
+
+      expect(async () => insertBooksLanguages(booksLanguagesArgs)).rejects.toEqual(new Error(errorMessage));
+    });
+  });
+
+  describe('updateBooksLanguages for books', () => {
+    let booksLanguagesForBooksArgs;
+    beforeEach(() => {
+      booksLanguagesForBooksArgs = [
```
Comment (alichherawalla) [inline server/daos/tests/booksLanguages.test.js]: use mockData file here as well

File: server/middleware/gqlAuth/tests/index.test.js [hunk 9]
```diff
@@ -186,107 +186,107 @@ describe('gqlAuth tests', () => {
...
-        query: `query Address{
-          address(id:1){
-            id
-          }
-        }
-        `
-      },
-      headers: {}
-    };
+  // describe('isAuthenticated tests', () => {
```
Comment (alichherawalla) [inline server/middleware/gqlAuth/tests/index.test.js]: letts cleanup

Comment (alichherawalla) [reply server/middleware/gqlAuth/tests/index.test.js]: we don't want to merge commented code. Let's clean all of it

Comment (sagar-wednesday) [reply server/middleware/gqlAuth/tests/index.test.js]: roger

**Review by alichherawalla — APPROVED** [review]

**Pull request merged**